src/main/java/com/minecolonies/core/client/gui/WindowHutAllInventory.java

```diff
@@ -201,21 +207,23 @@ private void updateResources()
 
         for (final BlockPos blockPos : containerList)
         {
-            final BlockEntity rack = world.getBlockEntity(blockPos);
-            if (rack instanceof TileEntityRack)
-            {
-                final Map<ItemStorage, Integer> rackStorage = ((TileEntityRack) rack).getAllContent();
+            final BlockEntity entity = world.getBlockEntity(blockPos);
+            Optional<IStorageBlockInterface> storageInterface = ModStorageBlocks.getStorageBlockInterface(entity);
+            if (storageInterface.isEmpty()) {
+                continue;
+            }
+
+            final Map<ItemStorage, Integer> storage = storageInterface.get().getAllContent();
```

Comment:
Same point as above, either use nullables or use the optional chaining

Reply:
Done.